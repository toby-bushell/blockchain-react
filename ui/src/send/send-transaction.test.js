import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import TransactionSend from './send-transaction';

const fs = require('fs');

describe('Transaction Component', () => {
  it('renders correctly', () => {
    const address = 'testAddress';
    const tree = renderer.create(<TransactionSend address={address} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders an address input', () => {
    const address = 'testAddress';
    expect(shallow(<TransactionSend address={address} />).find('input[name="toAddress"]').length).toEqual(1);
  });

  it('renders an amount input field', () => {
    const address = 'testAddress';
    expect(shallow(<TransactionSend address={address} />).find('input[name="amount"]').length).toEqual(1);
  });

  it('sets local error if form submit is empty', () => {
    const address = 'testAddress';

    const wrapper = shallow(<TransactionSend address={address} />);
    wrapper.find('form').simulate('submit', {
      target: {
        value: 'Change function'
      },
      preventDefault() {}
    });

    expect(wrapper.state('errors')).toEqual(expect.anything());
  });

  it('allows inputs saves to state and submits', () => {
    const address = 'testAddress';
    const wrapper = shallow(<TransactionSend address={address} />);

    // Update address and check state reflects
    wrapper.find('input[name="toAddress"]').simulate('change', {
      target: {
        value: 'newValue'
      },
      field: 'toAddress'
    });
    expect(wrapper.state().form.toAddress).toEqual(expect.anything());

    // Update amount and check state reflects
    wrapper.find('input[name="amount"]').simulate('change', {
      target: {
        value: 'newValue'
      },
      field: 'amount'
    });
    expect(wrapper.state().form.amount).toEqual(expect.anything());

    wrapper.instance().sendTransactionToApi = jest.fn(() => {
      console.log('\x1b[32m', 'jest function being called???', '\x1b[0m');
    });
    wrapper.update();

    // console.log(
    //   '\x1b[31m',
    //   'wrapper?',
    //   wrapper.sendTransactionToApi,
    //   wrapper,
    //   wrapper.instance().sendTransactionToApi,
    //   '\x1b[0m'
    // );
    // Check on submit the create transaction is called
    wrapper.find('form').simulate('submit', {
      target: {
        value: 'Change function'
      },
      preventDefault() {}
    });

    expect(wrapper.instance().sendTransactionToApi).toBeCalled();

    // const spy = jest.spyOn(TransactionSend.prototype, 'sendTransactionToApi');
    // wrapper.update();

    // expect(spy).toHaveBeenCalled();
    // expect(sendTransaction).toEqual(expect.anything());
  });
});
// it('renders correctly', () => {

//   // manually trigger the callback
//   tree.props.onSubmit();
//   // re-rendering
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });
