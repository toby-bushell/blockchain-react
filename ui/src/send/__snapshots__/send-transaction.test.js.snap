// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction Component renders correctly 1`] = `
<div>
  <h1>
    Transaction Send
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label
      className="label-hidden"
      htmlFor="amount"
    >
      Address to send to
    </label>
    <input
      name="toAddress"
      onChange={[Function]}
      placeholder="to address"
      type="text"
      value=""
    />
    <label
      className="label-hidden"
      htmlFor="amount"
    >
      Amount to send
    </label>
    <input
      name="amount"
      onChange={[Function]}
      placeholder="amount"
      type="number"
      value=""
    />
    <input
      className="sc-ifAKCX kMYCEa"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
